src/main/java/it/smartcommunitylab/aac/spid/provider/SpidIdentityProviderConfig.java

```diff
@@ -94,6 +97,50 @@ private SpidIdentityProviderConfig() {
         super();
     }
 
+    public void setBaseUrl(String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    public SpidIdentityProviderStatusMap getStatusMap() {
+        if (statusMap == null) {
+            statusMap = new SpidIdentityProviderStatusMap();
+            statusMap.setMetadataUrl(getMetadataUrl());
+            statusMap.setAssertionConsumerUrl(getAssertionConsumerUrl());
+        }
+        return statusMap;
+    }
+
+    public String getMetadataUrl() {
+        if (baseUrl != null) {
+            UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(metadataUrlTemplate());
+            return builder.buildAndExpand(Map.of("baseUrl", baseUrl, "metadataRegistrationId", getMetadataRegistrationId())).toUriString();
+        }
+        return null;
+    }
+
+    public String metadataUrlTemplate() {
+        return "{baseUrl}/auth/" + getAuthority() + "/metadata/{metadataRegistrationId}";
+    }
+
+    public String getAssertionConsumerUrl() {
+        if (baseUrl != null) {
+            UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(assertionConsumerUrlTemplate());
+            return builder.buildAndExpand(Map.of("baseUrl", baseUrl, "registrationId", getProvider())).toUriString();
```

Comment:
is the registration id the same for all spid registrations? I think we expose a single registration id for every pair <provider|upstream>.